EyeSeeTea/glass-dev
# Add logs amc calculations using d2-logger new library

### :pushpin: References

-   **Issue:** Closes [Add logger to AMC calculations](https://app.clickup.com/t/8693ecp1v)

### :memo: Implementation

- [x] Setup logger in GLASS app and in AMC script
- [x] Setup logger in AMC calculations tests
- [x] Add logger messages in AMC calculations and recalculations process
- [ ] Add [@eyeseetea/d2-logger](https://github.com/EyeSeeTea/d2-logger) new library (waiting for PR and publish first version https://github.com/EyeSeeTea/d2-logger/pull/1)
- [ ] Add configuration of logger with the program and data elements needed.
- [ ] Setting organization for logger

### :video_camera: Screenshots/Screen capture

### :fire: Testing
1. Change current ATC version in datastore glass --> ATCs 
2. In datastore have in glass --> amc-recalculation:
```
{
  "date": "2024-01-19T09:54:04.091Z",
  "orgUnitsIds": [
    "vboedbUs1As"
  ],
  "periods": [
    "2020"
  ],
  "recalculate": false
}
``` 
And change recalculate to true
3. Start script: yarn start-amc-recalculate --url "https://dev.eyeseetea.com/who-dev-238" --auth USERNAME:PASSWORD or upload file in GLASS app [AMC-SUBSTANCE-LEVEL-DATA-THA-TEMPLATE_short.xlsx](https://github.com/EyeSeeTea/glass-dev/files/14065666/AMC-SUBSTANCE-LEVEL-DATA-THA-TEMPLATE_short.xlsx)
4. After the script ends, check Capture program with the logs to see messages registered
